sql/pool.ts

```diff
@@ -0,0 +1,196 @@
+import type { SqlConnection, SqlConnectionOptions } from "./connection.ts";
+import type {
+  SqlClientQueriable,
+  SqlConnectableBase,
+  SqlPreparedStatement,
+  SqlQueryOptions,
+  SqlTransaction,
+  SqlTransactionOptions,
+} from "./core.ts";
+import type {
+  DeferredStack,
+  DeferredStackOptions,
+} from "../collections/deferred_stack.ts";
```

Comment:
This dependency feels strange to me. I assume `SqlClientPool` interface is agnostic with what is used for the implementation.

Reply:
I agree, it is better to have this agnostic